packages/@aws-cdk/aws-codecommit/test/codecommit.test.ts

```diff
@@ -66,6 +69,88 @@ describe('codecommit', () => {
 
     });
 
+    test('Repository can be initialized with contents from a ZIP file', () => {
+      // GIVEN
+      const app = new App();
+      const stack = new Stack(app, 'MyStack');
+
+      // WHEN
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromZipFile(join(__dirname, 'asset-test.zip')),
+      });
+
+      // THEN
+      const assetMetadata = app.synth().tryGetArtifact(stack.stackName)!.findMetadataByType(cxschema.ArtifactMetadataEntryType.ASSET);
+      expect(assetMetadata).toHaveLength(1);
+    });
+
+    test('Repository can be initialized with contents from a directory', () => {
+      // GIVEN
+      const app = new App();
+      const stack = new Stack(app, 'MyStack');
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromDirectory(join(__dirname, 'asset-test')),
+      });
+
+      // THEN
+      const assetMetadata = app.synth().tryGetArtifact(stack.stackName)!.findMetadataByType(cxschema.ArtifactMetadataEntryType.ASSET);
+      expect(assetMetadata).toHaveLength(1);
+    });
+
+    test('Repository can be initialized with contents from an asset', () => {
+      // GIVEN
+      const app = new App();
+      const stack = new Stack(app, 'MyStack');
+
+      const readmeAsset = new Asset(stack, 'ReadmeAsset', {
+        path: join(__dirname, 'asset-test'),
+      });
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromAsset(readmeAsset),
+      });
+
+      // THEN
+      const assetMetadata = app.synth().tryGetArtifact(stack.stackName)!.findMetadataByType(cxschema.ArtifactMetadataEntryType.ASSET);
+      expect(assetMetadata).toHaveLength(1);
+    });
+
+    test('Repository throws Error when initialized with file while expecting directory', () => {
+      // GIVEN
+      const app = new App();
+      const stack = new Stack(app, 'MyStack');
+
+      // THEN
+      expect(() => {
+        new Repository(stack, 'Repository', {
+          repositoryName: 'MyRepositoryName',
+          code: Code.fromDirectory(join(__dirname, 'asset-test/test.md')),
+        });
+      }).toThrow();
```

Comment:
Let's include the message checking too - right now, this will pass if the production code throws any exception, which is too wide:

```suggestion
      }).toThrow(/'asset-test/test.md' needs to be a path to a directory/);
```